auth/auth.c

```diff
@@ -449,41 +449,48 @@ static void gip_auth_exchange_rsa(struct work_struct *work)
 					     work_exchange_rsa);
 	struct gip_auth_pkt_host_secret pkt = {};
 	u8 random[GIP_AUTH_RANDOM_LEN * 2];
-	u8 pms[GIP_AUTH_SECRET_LEN];
+	u8 *pms;
 	int err;
 
+	pms = kmalloc(GIP_AUTH_SECRET_LEN, GFP_KERNEL);
+	if (!pms)
+		return -ENOMEM;
```

Comment:
Not sure why you don't see this but gip_auth_exchange_rsa doesn't return anything so this will cause build failure. Unclear what to do here, just call `dev_err(&auth->client->dev, "%s: allocation failed\n", __func__);` and return?

Reply:
Interesting that it didn't fail the build. In the make.log the is a warning about this but nothing more:
```
/var/lib/dkms/xone/v0.3-58-g5fff62d/build/auth/auth.c: In function ‘gip_auth_exchange_rsa’:
/var/lib/dkms/xone/v0.3-58-g5fff62d/build/auth/auth.c:457:10: warning: ‘return’ with a value, in function returning void
   return -ENOMEM;
          ^
/var/lib/dkms/xone/v0.3-58-g5fff62d/build/auth/auth.c:446:13: note: declared here
 static void gip_auth_exchange_rsa(struct work_struct *work)
             ^~~~~~~~~~~~~~~~~~~~~
```